peregrine/resources/submission/graphql/node.py

```diff
@@ -433,34 +433,43 @@ def resolve_node(self, info, **args):
             for node in psqlgraph.Node.get_subclasses()
             if node.label in subclasses_labels
         ]
+        q_all = query_with_args(subclasses, args, info)
     else:
-        subclasses = [psqlgraph.Node]
-
-    q_all = []
-    for subclass in subclasses:
-        q = query_with_args(subclass, args, info)
-        q_all.extend(q.all())
+        q_all = query_node_with_args(args, info)
```

Comment:
why not just use `q_all = query_with_args([psqlgraph.Node], args, info)`?

Reply:
Oh because `of_type` is now implemented in two different ways for `Node` and subclasses, and I didn't want to change their implementation within this ticket. The ancient `of_type` implementation was not working correctly with subclasses (the main issue fixed here), so I made `query_with_args` work under the same idea of yours to collect results from all selected tables and merge them in Python. And this is not compatible with the `Node` query.

Ideally, this whole logic could be rewritten with constructing a SQL unioning all relevant subqueries and do filtering once for all (fingers crossed for PostgreSQL's optimization). Let me create a new ticket for that under the performance scope.

Reply:
> And this is not compatible with the Node query.

For more detail, if we used only `query_with_args` like this:

```python
query_with_args([psqlgraph.Node], {'of_type': 'visit'}, info)
```

we will have `of_types = [Visit]`:

```python
    of_types = [psqlgraph.Node.get_subclass(label)
                for label in set(args.get('of_type', []))]
```

and fail to pass this condition check:

```python
        if not of_types or cls in of_types:
```

where `cls` is `psqlgraph.Node`, which is not in `[Visit]`.